src/controller/expenses.js

```diff
@@ -0,0 +1,140 @@
+'use strict';
+
+const expensesServices = require('../services/expensesServices');
+const usersServices = require('../services/usersServices');
+
+const getAll = (req, res) => {
+  const {
+    userId,
+    categories,
+    from,
+    to,
+  } = req.query;
+
+  let expenses = expensesServices.getAll();
+
+  if (userId) {
+    expenses = expenses
+      .filter(expense => expense.userId === +userId);
+  }
+
+  if (from) {
+    expenses = expenses.filter(expense => (
+      new Date(expense.spentAt) >= new Date(from))
+    );
+  }
+
+  if (to) {
+    expenses = expenses.filter(expense => (
+      new Date(expense.spentAt) <= new Date(to))
+    );
+  }
+
+  if (categories) {
+    expenses = expenses
+      .filter(expense => categories.includes(expense.category));
```

Comment:
if there are no `categories` in search params we will get `undefined`
if only 1 `categories` we will get just value (....?categories=1)
And if we have many categories we will get array of values (....?categories=1&categories=2&categories=3)

![image](https://github.com/mate-academy/node_accounting-app-v2/assets/86789161/ea010564-92a2-4116-9fcf-a66739d6d01f)

Reply:
Your code will crash if you have only 1 category